### PR TITLE
feat: use events to listen to token changes

### PR DESCRIPTION
**Pull request by pavanjoshi914**
### Describe the changes you have made in this PR

listen for token changes via alby-js-sdk events

### Type of change

(Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### How has this been tested?

trying oauth in extension

### Checklist

- [X] Self-review of changed code
- [X] Manual testing
- [X] Added automated tests where applicable
- [X] Update Docs & Guides
- For UI-related changes
- [X] Darkmode
- [X] Responsive layout

### CODE REVIEW

File: src/extension/background-script/connectors/alby.ts [hunk 1]
```diff
@@ -247,20 +247,21 @@ export default class Alby implements Connector {
...
-        } catch (error) {
-          // if auth token refresh fails, the refresh token has probably expired or is invalid
-          // the user will be asked to re-login
-          console.error("Failed to request new auth token", error);
-        }
+      if (!authClient.isAccessTokenExpired()) {
+        return authClient;
       }
 
+      authClient.on("tokenRefreshed", async (token: Token) => {
```
Comment (rolznz) [inline src/extension/background-script/connectors/alby.ts:254]: you need to register these listeners immediately after creating the authClient. Currently they won't be registered every time

File: src/extension/background-script/connectors/alby.ts [hunk 2]
```diff
@@ -247,20 +247,21 @@ export default class Alby implements Connector {
...
+        await this._updateOAuthToken(token);
+        return authClient;
```
Comment (rolznz) [inline src/extension/background-script/connectors/alby.ts:256]: Why return authClient here?

File: src/extension/background-script/connectors/alby.ts [hunk 3]
```diff
@@ -247,20 +247,21 @@ export default class Alby implements Connector {
         token: this.config.oAuthToken, // initialize with existing token
       });
 
-      if (this.config.oAuthToken) {
-        try {
-          if (authClient.isAccessTokenExpired()) {
-            const token = await authClient.refreshAccessToken();
```
Comment (rolznz) [inline src/extension/background-script/connectors/alby.ts:253]: this code needs to be re-added

**Comment by pavanjoshi914** [comment]
@rolznz any updates here?

File: src/extension/background-script/connectors/alby.ts [hunk 4]
```diff
@@ -247,11 +247,18 @@ export default class Alby implements Connector {
...
+      authClient.on("tokenRefreshFailed", async (error: Error) => {
+        console.error("Failed to Refresh token", error);
+      });
+
       if (this.config.oAuthToken) {
         try {
           if (authClient.isAccessTokenExpired()) {
-            const token = await authClient.refreshAccessToken();
-            await this._updateOAuthToken(token.token);
+            await authClient.refreshAccessToken();
```
Comment (rolznz) [inline src/extension/background-script/connectors/alby.ts:261]: @pavanjoshi914 can you also look at the _request function? we need changes there

**Comment by pavanjoshi914** [comment]
closing in favour of #2830 as there is lots of dependency conflict in yarn.lock

**Pull request closed without merging**